pkg/template/template.go

```diff
@@ -169,6 +169,10 @@ func (p *Processor) GenerateParameterValues(t *api.Template) error {
 				return fmt.Errorf("template.parameters[%v]: Unable to convert the generated value '%#v' to string", i, value)
 			}
 		}
+		if len(param.Value) == 0 && param.Required {
+			return fmt.Errorf("template.parameters[%v]: parameter is marked required but has no default value and no generator", i)
```

Comment:
do we have an example of this elsewhere?  I can create a new error struct but i'm not clear where that gets us in terms of manifesting it to the REST client.